### PR TITLE
Virtio Qemu 8.1.2 issues fix

### PR DESCRIPTION
**Pull request by CV-Bowen**
## Summary
@masayuki2009 noticed that rv-virt smp not work in QEMU-8.1.2, in detal the CPU3 stuck at an **atomic instruction when access an io address.** This patch fix this issue by using the nuttx implemented io operation to relplace the libmetal implemented io read/write operation to avoid use atomic operation to the io address.
This PR contains three commits:

Commit 1:
Modify the virtio-mmio.c to use to the nuttx implemented io operation.

Commit 2:
Add a patchset to openamp/libmetal to make nuttx io read/write operation width matched to access the io address when width = 1, 2, 4, 8.

Commit 3:
Enable config `CONFIG_DEV_SIMPLE_ADDRENV` for all the qemu configs that used virtio to avoid compile error. Becasue function `up_addrenv_pa_to_va()` and `up_addrenv_va_to_pa` are not implemented by default.

## Impact
Qemu configs that used virtio

## Testing
qemu-armv8a:netnsh, qemu-armv8a:netnsh_smp, rv-virt:netnsh, rv-virt:netnsh_smp

### CODE REVIEW

**Comment by CV-Bowen** [comment]
@masayuki2009 Could try this PR, I verified this PR can fix the rv-virt smp issue in QEMU-8.1.2.

**Comment by masayuki2009** [comment]
@CV-Bowen 
Let me try this PR tomorrow.

File: openamp/0002-libmetal-nuttx-io.c-align-access-when-read-write-siz.patch [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Signed-off-by: Bowen Wang <wangbowen6@xiaomi.com>
+---
+ lib/system/nuttx/io.c | 25 +++++++++++++++++++++++--
+ 1 file changed, 23 insertions(+), 2 deletions(-)
+
+diff --git a/lib/system/nuttx/io.c libmetal/lib/system/nuttx/io.c
+index 4fa4727..3ce9cbe 100644
+--- a/lib/system/nuttx/io.c
++++ libmetal/lib/system/nuttx/io.c
```
Comment (xiaoxiang781216) [inline openamp/0002-libmetal-nuttx-io.c-align-access-when-read-write-siz.patch:15]: let's upstream to open-amp community

**Review by jerpelea — APPROVED** [review]

File: drivers/virtio/virtio-mmio.c [hunk 2]
```diff
@@ -759,9 +759,9 @@ static int virtio_mmio_init_device(FAR struct virtio_mmio_device_s *vmdev,
   vmdev->shm_phy = (metal_phys_addr_t)0;
   vmdev->cfg_phy = (metal_phys_addr_t)regs;
   metal_io_init(&vmdev->shm_io, NULL, &vmdev->shm_phy,
-                SIZE_MAX, UINT_MAX, 0, NULL);
+                SIZE_MAX, UINT_MAX, 0, metal_io_get_ops());
```
Comment (xiaoxiang781216) [inline drivers/virtio/virtio-mmio.c:762]: why not change metal io operation to non-atomic directly?

**Comment by masayuki2009** [comment]
@CV-Bowen 
I confirmed that this PR fixes the issue with QEMU-8.1.2

**Review by masayuki2009 — APPROVED** [review]

**Pull request merged**